src/prefect/cli/deploy.py

```diff
@@ -428,6 +431,22 @@ async def _run_single_deploy(
             console=app.console, client=client
         )
 
+    triggers: List[DeploymentTrigger] = []
+    trigger_specs = deploy_config.get("triggers")
+    if trigger_specs:
+        triggers = _initialize_deployment_triggers(deployment_name, trigger_specs)
+        if client.server_type != ServerType.CLOUD:
```

Comment:
why is it that we don't check the serve type prior to initializing deployment triggers?

Reply:
My thinking was that we should still validate the triggers property even if we aren't going to, ultimately, create them.

Reply:
oh I see, it's just validating them 👍

Reply:
Right, it's initializing the pydantic model, which also validates.